### ISSUE TITLE
"TypeError: request.replace is not a function" when building project

### ISSUE THREAD

**Issue by unfalse** [webpack/webpack#5701]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
Error on running node script, which is executing webpack-dev-middleware: 
```
D:\MyExperiments\nnn-mmm-ui_npm_update\node_modules\webpack\lib\RequestShortener.js:15
        return request.replace(NORMALIZE_SLASH_DIRECTION_REGEXP, "/");
                       ^
TypeError: request.replace is not a function
    at normalizeBackSlashDirection (D:\MyExperiments\nnn-mmm-ui_npm_update\node_modules\webpack\lib\RequestShortener.js:15:17)

```
**If the current behavior is a bug, please provide the steps to reproduce.**
Execute `npm run node` where `node` is
`node --harmony --max_old_space_size=8192 --optimize_for_size --max_executable_size=4096 --stack_size=4096 app.js`
The code in app.js executed another script with following strings in it:
`clientApp.use(webpackMiddleware(webpack(webpackConfig),
                {
                    noInfo: false,
                    quiet: false,
                    stats: { colors: true }
                }));`
Webpack config and logs you can see [here in this gist](https://gist.github.com/unfalse/c810edf5847f2a84d26d18a00308113e) 

**What is the expected behavior?**
Project successfully builds without errors.

**If this is a feature request, what is motivation or use case for changing the behavior?**
**Please mention other relevant information such as the browser version, Node.js version, webpack version and Operating System.**
Browser is not playing any role here because everything happening only in command line.
Node.js is v6.11.1
Webpack is 3.5.6
OS is Windows 10, version 1703 (OS Build 15063.608)

**Comment by unfalse** [comment]
For anyone who interested: I downgraded webpack version to 2 and it started to output clear and useful errors which is easy to fix.

**Comment by unfalse** [comment]
Finally I got it working.
Firstly, webpack is launching from a script in a project's root folder and the config resides in a sub-folder.
Because of that, dependencies is taken from root's packages.json.
I downgraded the webpack version and launched it from a command line using gitbash (from a subfolder) like that
`./node_modules/webpack/bin/webpack.js` --config ./webpack.config.babel.js --display-error-details`
Webpack shown many errors which I was able to fix. When I finished with fixing, I launched webpack from a script and it worked fine.
One important detail that I forgot to mention before: the project is on Angular and it needed raw-loader and not html-loader. Also it needed options for stylus-loader
`stylus-loader?resolve url`
